src/game/savegame_bson.c

```diff
@@ -163,95 +163,114 @@ static struct json_value_s *Savegame_BSON_ParseFromFile(MYFILE *fp)
-        start->flags.got_uzis = json_object_get_bool(start_obj, "got_uzis", 0);
-        start->flags.got_shotgun =
-            json_object_get_bool(start_obj, "got_shotgun", 0);
-        start->flags.costume = json_object_get_bool(start_obj, "costume", 0);
+        RESUME_INFO *resume = &resume_info[i];
+        resume->lara_hitpoints = json_object_get_int(
+            resume_obj, "lara_hitpoints", g_Config.start_lara_hitpoints);
+        resume->pistol_ammo = json_object_get_int(resume_obj, "pistol_ammo", 0);
+        resume->magnum_ammo = json_object_get_int(resume_obj, "magnum_ammo", 0);
+        resume->uzi_ammo = json_object_get_int(resume_obj, "uzi_ammo", 0);
+        resume->shotgun_ammo =
+            json_object_get_int(resume_obj, "shotgun_ammo", 0);
+        resume->num_medis = json_object_get_int(resume_obj, "num_medis", 0);
+        resume->num_big_medis =
+            json_object_get_int(resume_obj, "num_big_medis", 0);
+        resume->num_scions = json_object_get_int(resume_obj, "num_scions", 0);
+        resume->gun_status = json_object_get_int(resume_obj, "gun_status", 0);
+        resume->gun_type = json_object_get_int(resume_obj, "gun_type", 0);
+        resume->flags.available =
+            json_object_get_bool(resume_obj, "available", 0);
+        resume->flags.got_pistols =
+            json_object_get_bool(resume_obj, "got_pistols", 0);
+        resume->flags.got_magnums =
+            json_object_get_bool(resume_obj, "got_magnums", 0);
+        resume->flags.got_uzis =
+            json_object_get_bool(resume_obj, "got_uzis", 0);
+        resume->flags.got_shotgun =
+            json_object_get_bool(resume_obj, "got_shotgun", 0);
+        resume->flags.costume = json_object_get_bool(resume_obj, "costume", 0);
+
+        resume->stats.timer =
+            json_object_get_int(resume_obj, "timer", resume->stats.timer);
+        resume->stats.secret_flags = json_object_get_int(
+            resume_obj, "secrets", resume->stats.secret_flags);
+        resume->stats.kill_count =
+            json_object_get_int(resume_obj, "kills", resume->stats.kill_count);
+        resume->stats.pickup_count = json_object_get_int(
+            resume_obj, "pickups", resume->stats.pickup_count);
+        resume->stats.death_count = json_object_get_int(
+            resume_obj, "deaths", resume->stats.death_count);
+        resume->stats.max_secret_count = json_object_get_int(
+            resume_obj, "max_secrets", resume->stats.max_secret_count);
+        resume->stats.max_kill_count = json_object_get_int(
+            resume_obj, "max_kills", resume->stats.max_kill_count);
+        resume->stats.max_pickup_count = json_object_get_int(
+            resume_obj, "max_pickups", resume->stats.max_pickup_count);
     }
     return true;
 }
 
-static bool Savegame_BSON_LoadEndInfo(
-    struct json_array_s *end_arr, GAME_INFO *game_info)
+static bool Savegame_BSON_LoadCurrentInfo(
+    struct json_array_s *current_arr, GAME_INFO *game_info)
```

Comment:
Let's rename or comment that this function exists solely for backwards compatibility with 2.6 and 2.7 saves.